flang/test/CMakeLists.txt

```diff
@@ -64,6 +64,7 @@ set(FLANG_TEST_DEPENDS
   FortranRuntime
   Fortran_main
   FortranDecimal
+  clang
```

Comment:
Is this pulling in all of `clang`? Do we really need that? If this is only for testing, can we use `REQUIRES: clang` in the test to make sure clang is available?

Reply:
Sounds good.  I'll remove it.

Reply:
`REQUIRES: clang` only affects whether the LIT test is run or not.  The `clang` keyword, in this case, has to be present in `config.available_features` for the test to run.  Since we do not add `clang` into the available features list, the test will just not run.  Even if we did enabled `clang` as a LIT feature, we still have to guarantee that the `clang` target (that builds the actual `clang` executable) is built before the test runs - this can only be done by setting proper dependency in CMake files.

If building the whole `clang` under `check-flang` is not acceptable, we need another solution.

Reply:
I generally use `check-flang` to build flang. Also, was not sure whether to add a dependency with the whole of clang for just one test.

Yes, we have to add `clang` to `config.available_features` for REQUIRES to work. 

For the dependency with clang, Do we really want to check with the `clang` just built or is it sufficient to check with the clang installed on the system? Also, is it OK to check with the system compiler?

The OpenMP runtime library checks for the presence of `flang` for running an offload test. Would that be a possible way?
https://reviews.llvm.org/D158546

Another reference might include the way the runtime no dependency on C++ libs is checked.
https://reviews.llvm.org/D104290

Reply:
> For the dependency with clang, Do we really want to check with the clang just built or is it sufficient to check with the clang installed on the system? Also, is it OK to check with the system compiler?

Yes, we need to check with the `clang` that has just been built.   The test is checking that `clang` is able to find the header file, so `clang` used for the test is the one which has the header file in its "installation" structure.

If we change the LIT test from `clang` compilation +`flang` compilation + execution into just `clang` compilation, then we can move this functionality into `clang/lib/Headers` completely.

We can have an end-to-end test with `clang`/`flang`/executable in llvm-test-suite as you suggested.

I think adding `clang` dependency is too much overhead.

Reply:
Moving the functionality into clang would make clang dependent on flang, which we don't want.  I **do** think we want to keep the file ISO_Fortran_binding.h in the flang project.

Reply:
I do not think the "dependency" I am talking about is a problem.  If we move the LIT tests into clang source tree, and do not use `flang-new` in the test, then the only dependency is the ability for `clang` CMake code to access `ISO_Fortran_binding.h`, which is already a part of the LLVM source tree.  So it should not be a problem.

Anyway, I am okay with keeping it as-is, but without having the `clang` dependency for `check-flang`.  Moving the LIT test out of the tree should "resolve" this.

Reply:
Is there a way to make this test configurable based on a cmake variable?  That way, we could keep in in the flang tree and keep it so that it created and executes a program.

Reply:
After talking to Slava.  This doesn't seem like a viable alternative.